src/MIDebugEngine/AD7.Impl/AD7Disassembly.cs

```diff
@@ -162,6 +173,64 @@ public int Read(uint dwInstructions, enum_DISASSEMBLY_STREAM_FIELDS dwFields, ou
                     }
                 }
 
+                if (idx == 0 && m_lastSourceInfoStale)
+                {
+                    m_lastSourceInfoStale = false;
+
+                    ulong previousInstructionAddress = instruction.Addr - 1;
+
+                    if (previousInstructionAddress >= 0 && previousInstructionAddress < (ulong)prgDisassembly.Length)
+                    {
+                        DisassemblyData previousInstruction = prgDisassembly[previousInstructionAddress];
```

Comment:
>previousInstructionAddress [](http://example.com/codeflow?start=77&length=26)

I am not following this logic. `prgDisassembly` is an array. So you can't used an instruction address here.

Instead, what I think you want is:
* We need to SeekBack an extra instruction, so:
   * Add a new bool field to know if we have already seaked back an extra instruction (something like `m_skipNextInstruction`)
   * In SeekBack, if the last instruction data is stale, SeekBack an extra instruction and set the m_skipNextInstruction to true
   * In the start of this method, if the previous instruction data is stale and m_skipNextInstruction is false, call `SeekBack(1)`.
   * In this method, if `m_skipNextInstruction` is true then we need to increment the requested number of instructions.
   * In this method, if `m_skipNextInstruction` is true then we need to:
      * Do a `.FirstOrDefault()` to process the first instruction and fill in the previous instruction data
      * Do a `.Skip(1)` on the instruction enumerator